File: 0040-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  let lo = 0,
    hi = nums.length;
  while (lo < hi) {
    const m = Math.floor(lo + (hi - lo) / 2);
    if (nums[m] === target) return m;
    if (nums[m] > target) hi = m;
    if (nums[m] < target) lo = m + 1;
  }
  return -1;
};

File: 0040-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int leftIdx = 0;
        int rightIdx = nums.length;

        while (leftIdx < rightIdx) {
            int midIdx = leftIdx + (rightIdx - leftIdx) / 2;
            int midItem = nums[midIdx];

            if (target == midItem) {
                return midIdx;
            }

            if (target < midItem) {
                rightIdx = midIdx;
            }

            if (target > midItem) {
                leftIdx = midIdx + 1;
            }
        }

        return -1;
    }
}